### PR TITLE
#166789958 Users Should Be Able to Comment on Articles

### PR DESCRIPTION
**Pull request by Hervera**
#### What does this PR do?
Enables an authenticated user comment on articles

#### Description of Task to be completed
 - Add comment component
 - Connect comment to article component
 - Add comment functionality
 - Add tests

#### How should this be manually tested?
1. Run the following commands in the terminal
```
Clone the project
 cd ah-92explorers-frontend
 git checkout ft-comment-on-articles-166789958
`npm i` to install all dependencies 
`npm run dev` and navigate to localhost:8080/article/:articleSlug/ to make a comment
```

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166789958](https://www.pivotaltracker.com/story/show/166789958)

#### Screenshots (if appropriate)
Screen Shot 2019-09-09 at 01 46 16

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

File: src/redux/actions/actionCreators/login.js
Comment (elemanhillary-zz) [inline src/redux/actions/actionCreators/login.js:42]: you call the backend endpoint for logout so that not only the token is cleared from the backend but also be blacklisted, so that it wont be re-used in case someone logs out and stores the token somewhere and hardcodes it back to the `localStorage`

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]

File: src/redux/actions/actionCreators/login.js
Comment (Hervera) [reply src/redux/actions/actionCreators/login.js:42]: I have made that change when someone logout, it clears everything in localStorage.

**Review by niyoceles — DISMISSED** [review]
It can be merged after merge `reading an article feature`.

**Review by kodek-sleuth — DISMISSED** [review]

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]
I have realised that you are using `toastify` for error and success messages, I think we all agreed to use The messages component. Please make use of this team convention. While at that please also change the comment cards background from `#fff` to `#eeeee` as we had earlier discussed. This is ready to be merged after those changes have been exerted.

**Comment by IsaiahRn** [comment]
> I have realised that you are using `toastify` for error and success messages, I think we all agreed to use The messages component. Please make use of this team convention. While at that please also change the comment cards background from `#fff` to `#eeeee` as we had earlier discussed. This is ready to be merged after those changes have been exerted.

@kodek-sleuth For the kind of messages that are being returned from the messages component are somehow static, but operations like commenting, following, liking, rating, i think that we can use `toasify` because it provides nice messages and they are dynamic.

**Comment by Hervera** [comment]
> I have realised that you are using `toastify` for error and success messages, I think we all agreed to use The messages component. Please make use of this team convention. While at that please also change the comment cards background from `#fff` to `#eeeee` as we had earlier discussed. This is ready to be merged after those changes have been exerted.

@kodek-sleuth `toastify` is dynamic while `message component` is static. `toastify` pops up a message to show a user that has commented, deleted a comment and other things without scrolling up or down the page to see the`message component` which is static somewhere on the page. For showing direct messages to a user while commenting, following, liking and rating. we are going to use `toastify` instead of `message component`.  For comment cards background from #fff to #eeeee, this can be changed.

**Review by IsaiahRn — APPROVED** [review]
A user can be able to comment on an article, it is ready for merge!

**Review by kodek-sleuth — APPROVED** [review]

**Review by niyoceles — APPROVED** [review]

**Review by elemanhillary-zz — APPROVED** [review]
ready for review

**Pull request merged**
